http/middleware.go

```diff
@@ -127,20 +148,20 @@ type LogData struct {
 	Panic              any
 }
 
-// WithData returns logger with filled fields.
-func WithData(l logger.ServerLogger, ld LogData) logger.ServerLogger {
+// WithData returns slog with filled fields.
+func WithData(l *slog.Logger, ld LogData) *slog.Logger {
```

Comment:
I think this would be better if `LogData` struct implemented the `slog.LogValuer` interface and generated its own log representation, than doing it here like this. It would make `LogData` loggable consistently regardless of where or how it is sent to the logs.

Reply:
That's a good point @robertrossmann! Will look into it!